modules/security/src/main/IpIntel.scala

```diff
@@ -11,36 +12,45 @@ final class IpIntel(
     contactEmail: EmailAddress
 )(implicit ec: scala.concurrent.ExecutionContext) {
 
-  def apply(ip: IpAddress): Fu[Int] = failable(ip).nevermind
+  def apply(ip: IpAddress, reason: IpIntel.Reason): Fu[Int] = failable(ip, reason).nevermind
 
-  def failable(ip: IpAddress): Fu[Int] =
+  def failable(ip: IpAddress, reason: IpIntel.Reason): Fu[Int] =
     if (IpIntel isBlacklisted ip) fuccess(90)
     else if (contactEmail.value.isEmpty) fuccess(0)
-    else cache get ip
-
-  private val cache = cacheApi[IpAddress, Int](8192, "ipIntel") {
-    _.expireAfterWrite(3 days)
-      .buildAsyncFuture { ip =>
```

Comment:
For review: I am least confident about the change from `AsyncLoadingCache` to `AsyncCache`.